### PR TITLE
Change tenant mgt tenant delete function to invoke post delete methods.

### PR DESCRIPTION
**Pull request by bhagyasakalanka**

### CODE REVIEW

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java [hunk 1]
```diff
@@ -774,4 +779,22 @@ private static boolean isFormatCorrect(String regularExpression, String domainNa
         Matcher m2 = p2.matcher(domainName);
         return m2.matches();
     }
+
+    /**
+     * Triggers post tenant delete for TenantMgtListener
+     *
+     * @param tenantUuid String tenant unique identifier
+     * @param userUuid String user unique identifier
```
Comment (somindatommy) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:787]: Lets mention this is admin user uuid

Comment (bhagyasakalanka) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java]: fixed 45acae978b872ecc6afaa3061417abb6164da3e0

**Review by somindatommy — APPROVED** [review]

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java [hunk 2]
```diff
@@ -774,4 +773,23 @@ private static boolean isFormatCorrect(String regularExpression, String domainNa
...
+     * @param tenantId int Tenant id.
+     * @param tenantUuid String Tenant unique identifier.
+     * @param adminUserUuid String Tenant admin user unique identifier.
+     * @throws StratosException  If trigger failed.
+     */
+    private static void triggerPostTenantDelete(int tenantId, String tenantUuid, String adminUserUuid)
+            throws StratosException {
+
+        for (TenantMgtListener tenantMgtListener : TenantMgtServiceComponent
+                .getTenantMgtListeners()) {
```
Comment (AnuradhaSK) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:789]: Formatting issue. This line can be move into the previous line

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java [hunk 3]
```diff
@@ -774,4 +773,23 @@ private static boolean isFormatCorrect(String regularExpression, String domainNa
...
+            log.debug("Executing OnPostDelete on Listener Impl Class Name : "
```
Comment (AnuradhaSK) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:790]: check `if (log.isDebugEnabled())` before printing debug logs

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java [hunk 4]
```diff
@@ -774,4 +773,23 @@ private static boolean isFormatCorrect(String regularExpression, String domainNa
...
         return m2.matches();
     }
+
+    /**
+     * Triggers post tenant delete for TenantMgtListener.
+     *
+     * @param tenantId int Tenant id.
+     * @param tenantUuid String Tenant unique identifier.
+     * @param adminUserUuid String Tenant admin user unique identifier.
+     * @throws StratosException  If trigger failed.
```
Comment (AnuradhaSK) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:783]: Foramatting issue

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java (hunk 2 above)
Comment (bhagyasakalanka) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:789]: Fixed in https://github.com/wso2/carbon-multitenancy/pull/218

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java (hunk 3 above)
Comment (bhagyasakalanka) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:790]: Fixed in https://github.com/wso2/carbon-multitenancy/pull/218

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java (hunk 4 above)
Comment (bhagyasakalanka) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:783]: fixed in https://github.com/wso2/carbon-multitenancy/pull/219

**Pull request merged**